internal/services/report_processor.go

```diff
@@ -174,71 +174,69 @@ func ProcessReport(msg *kafka.Message) {
 				}
 			}
 
-			// Below we make sure that report which is been processed is the latest(interval_endtime) report.
-			// If not then replace the maxEndTime so that we can recommendation is again calculated for the same(latest)
-			// interval_endtime
-			if recommendation_stored_in_db, err := model.GetFirstRecommendationSetsByWorkloadID(workload.ID); err != nil {
+			recommendation_stored_in_db, err := model.GetFirstRecommendationSetsByWorkloadID(workload.ID)
+			if err != nil {
 				log.Errorf("Error while checking for recommendation_set record: %s", err)
 				continue
-			} else if !reflect.ValueOf(recommendation_stored_in_db).IsZero() {
-				if recommendation_stored_in_db.MonitoringEndTime.UTC().After(maxEndTime) {
-					maxEndTime = recommendation_stored_in_db.MonitoringEndTime.UTC()
-				}
 			}
 
-			recommendation, err := kruize.Update_recommendations(experiment_name, maxEndTime)
-			if err != nil {
-				end_interval := utils.ConvertDateToISO8601(maxEndTime.String())
-				if err.Error() == fmt.Sprintf("Recommendation for timestamp - \" %s \" does not exist", end_interval) {
-					log.Infof("Recommendation does not exist for timestamp - \" %s \"", end_interval)
+			maxEndtimeFromDB := recommendation_stored_in_db.MonitoringEndTime.UTC()
+			maxEndtimeFromReport := maxEndTime.UTC()
+			duration := maxEndtimeFromReport.Sub(maxEndtimeFromDB)
+
+			if reflect.ValueOf(recommendation_stored_in_db).IsZero() {
```

Comment:
Does this mean when there are no recommendations for workload in db without checking if it has passed 24 hours we try to get recommendation for that? If yes then this should also be happening after 24 hours?